src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php

```diff
@@ -0,0 +1,211 @@
+        if (null === $this->dbPassword) {
+            $this->dbPassword = \ChameleonSystem\CoreBundle\ServiceLocator::getParameter('database_password');
+        }
+    }
+
+    public function __destruct()
+    {
+        if (null !== $this->dbConnection) {
+            $this->dbConnection->close();
+        }
+    }
+
+    /**
+     * returns true if the init was ok - else false.
+     *
+     * @param $sTable
+     * @param $aFields
+     *
+     * @return bool
+     */
+    public function Initialize($sTable, $aFields)
+    {
+        $this->sTableName = $sTable;
+        $this->aFields = $aFields;
+        $this->sFileName = CMS_TMP_DIR.'/cms_bulk_'.$this->sTableName;
+        $this->rFile = fopen($this->sFileName, 'wb');
+        if (false !== $this->rFile) {
+            fclose($this->rFile);
+            $this->rFile = fopen($this->sFileName, 'wb');
+        }
+
+        return false !== $this->rFile;
+    }
+
+    /**
+     * return true if the data was writ ten to the target file.
+     *
+     * @param $aData
+     *
+     * @return bool
+     */
+    public function AddRow($aData)
+    {
+        if (false === $this->rFile) {
+            return false;
+        }
+        $aDataEscaped = $this->EscapeData($aData);
+        $sRow = "'".implode("'\t'", $aDataEscaped)."'\n";
+        $rReturn = fwrite($this->rFile, $sRow);
+
+        return false !== $rReturn;
+    }
+
+    public function CommitData()
+    {
+        fclose($this->rFile);
+
+        $databaseConnection = $this->getDbConnection();
+        $quotedTableName = $databaseConnection->real_escape_string($this->sTableName);
+        $quotedFields = array_map(array($databaseConnection, 'real_escape_string'), $this->aFields);
+        $query = "LOAD DATA LOCAL INFILE '{$this->sFileName}'
+                         REPLACE
+                      INTO TABLE `$quotedTableName`
+                   CHARACTER SET 'utf8'
+                          FIELDS TERMINATED BY '\\t' ENCLOSED BY '\\''
+                           LINES TERMINATED BY '\\n' STARTING BY ''
+                                 (`".implode('`,`', $quotedFields).'`)
+               ';
+        $databaseConnection->query($query);
+        $sError = $databaseConnection->error;
+        if (empty($sError)) {
+            unlink($this->sFileName);
+
+            return true;
+        } else {
+            trigger_error('MySQL error: "'.$sError.'" in query '.$query.' - switching to backup import via shell call', E_USER_WARNING);
+            $this->CommitDataViaShell($query);
+
+            return true;
+        }
+    }
+
+    /**
+     * in some cases (probably due to php bug #55737) php is not able to perform a LOAD DATA
+     * LOCAL INFILE using a non-root mysql user. In this case we'll fall back to an import
+     * call via shell. Be sure the user is allowed to perform a LOAD DATA LOCAL INFILE command.
+     *
+     * @param string $sQuery
+     *
+     * @return mixed (0 on success)
```

Comment:
Based on the implementation, the return type should probably be `int|false`. The important distinction being that `false` and non-0 denotes an error while 0 denotes success